src/Plugin/QueueWorker/EntityQueueWorker.php

```diff
@@ -50,6 +57,11 @@ public function processItem($data): void {
 
     if (empty($data['delete'])) {
       $entity = $storage->load($data['entity_id']);
+      if ($this->seenStore->setIfNotExists($entity->id(), 1)) {
```

Comment:
@rkunjappan We only run the insert hook for a document if it's new (i.e. we haven't "seen" it before).

Reply:
Thank you for confirming @chx